packages/core/src/html-entrypoint.ts

```diff
@@ -105,19 +105,19 @@ export class HTMLEntrypoint {
             if (!base) {
               // this bundle only exists in the fastboot variant
               let element = placeholder.start.ownerDocument.createElement('fastboot-script');
-              element.setAttribute('src', this.publicAssetURL + fastboot);
+              element.setAttribute('src', this.publicAssetURL + relativeToAssetDir(fastboot as string));
               placeholder.insert(element);
               placeholder.insertNewline();
             } else if (!fastboot || base === fastboot) {
               // no specialized fastboot variant
-              let src = this.publicAssetURL + base;
+              let src = this.publicAssetURL + relativeToAssetDir(base);
               placeholder.insertURL(src);
             } else {
               // we have both and they differ
-              let src = this.publicAssetURL + base;
+              let src = this.publicAssetURL + relativeToAssetDir(base);
               let element = placeholder.insertURL(src);
               if (element) {
-                element.setAttribute('data-fastboot-src', this.publicAssetURL + fastboot);
+                element.setAttribute('data-fastboot-src', this.publicAssetURL + relativeToAssetDir(fastboot));
```

Comment:
Having to repeat this logic everywhere we want to generate an asset url with `this.publicAssetURL + something` is messy but I've elected to submit for discussion as is before adding any indirection/abstraction.

If we're happy with this proposed approach, perhaps we could make it cleaner by adding something like `generateAssetURL` to hide this logic away:

```suggestion
                element.setAttribute('data-fastboot-src', generateAssetURL(this.publicAssetURL, fastboot));
```